### PR TITLE
Removed usage of AccessibilityDebugLogger from public API

### PR DESCRIPTION
**Pull request by ASalavei**
Fixes: https://youtrack.jetbrains.com/issue/CMP-6767/Remove-AccessibilityDebugLogger-from-public

### Breaking Changes - iOS
- Update `AccessibilitySyncOptions` and remove `AccessibilityDebugLogger` from public API

### CODE REVIEW

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/uikit/ComposeUIViewControllerConfiguration.uikit.kt [hunk 1]
```diff
@@ -103,15 +104,11 @@ sealed interface OnFocusBehavior {
     /**
      * The Compose view will stay on the current position.
      */
-    object DoNothing : OnFocusBehavior
+    data object DoNothing : OnFocusBehavior
```
Comment (MatkovIvan) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/uikit/ComposeUIViewControllerConfiguration.uikit.kt:107]: [Avoid using data classes in your API﻿](https://kotlinlang.org/docs/api-guidelines-backward-compatibility.html#avoid-using-data-classes-in-your-api)

Comment (ASalavei) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/uikit/ComposeUIViewControllerConfiguration.uikit.kt:107]: These are data objects, not classes, so they cannot be changed in the way it described in the article.

**Review by MatkovIvan — APPROVED** [review]
Requires second approval before merge

**Review by elijah-semyonov — APPROVED** [review]

**Comment by igordmn** [comment]
Changed to "Breaking changes". Though, this change is only for alpha01, not for Stable, as we removed it in another PR

**Pull request merged**